TCAT/Cells/RouteTableViewCell.swift

```diff
@@ -113,124 +65,124 @@ class RouteTableViewCell: UITableViewCell {
-            containerView.topAnchor.constraint(equalTo: contentView.topAnchor),
-            containerView.bottomAnchor.constraint(equalTo: contentView.bottomAnchor)
-        ])
-
-        NSLayoutConstraint.activate([
-            verticalStackView.leadingAnchor.constraint(equalTo: containerView.leadingAnchor),
-            verticalStackView.trailingAnchor.constraint(equalTo: containerView.trailingAnchor),
-            verticalStackView.topAnchor.constraint(equalTo: containerView.topAnchor),
-            verticalStackView.bottomAnchor.constraint(equalTo: containerView.bottomAnchor)
-        ])
-
-        NSLayoutConstraint.activate([
-            arrowImageView.heightAnchor.constraint(equalToConstant: arrowImageViewHeight),
-            arrowImageView.widthAnchor.constraint(equalToConstant: arrowImageViewWidth),
-            arrowImageView.centerYAnchor.constraint(equalTo: departureTimeLabel.centerYAnchor)
-        ])
+    private func setLiveIndicatorViewsConstraints() {
+        let spaceBtnLiveElements: CGFloat = 4
+
+        liveLabel.snp.remakeConstraints { make in
+            make.leading.top.equalToSuperview()
+            make.bottom.equalToSuperview()
+        }
+
+        liveIndicatorView.snp.remakeConstraints { make in
+            make.leading.equalTo(liveLabel.snp.trailing).offset(spaceBtnLiveElements)
+            make.centerY.equalTo(liveLabel)
+        }
 
     }
 
-    private func setTranslatesAutoresizingMaskIntoConstraints() {
-        let subviews = [timesStackView, travelTimeLabel,
-                        departureStackView, departureTimeLabel, arrowImageView,
-                        liveStackView, liveLabel, liveIndicatorView, stretchyFillerView,
-                        verticalStackView, routeDiagram, containerView]
-        subviews.forEach { $0.translatesAutoresizingMaskIntoConstraints = false }
+    private func setupDataDependentConstraints() {
+        let routeDiagramTopOffset = 8
+
+        liveContainerView.snp.makeConstraints { make in
+            make.bottom.equalTo(routeDiagram.snp.top).offset(-routeDiagramTopOffset)
+        }
+
+        // Set trailing and bottom prioirites to 999 to surpress constraint errors
+        routeDiagram.snp.makeConstraints { make in
+            make.top.equalTo(liveContainerView.snp.bottom).offset(routeDiagramTopOffset)
+            make.leading.equalTo(travelTimeLabel)
+            make.trailing.equalTo(departureStackView).priority(999)
+            make.bottom.equalToSuperview().inset(containerViewLayoutInsets).priority(999)
+        }
     }
 
-    /// For debugging constraint errors
-    private func setDebugIdentifiers() {
-        timesStackView.accessibilityIdentifier = "timesStackView"
-        travelTimeLabel.accessibilityIdentifier = "travelTimeLabel"
+    // MARK: Set Data
+    func configure(for route: Route, delegate: (UIViewController & RouteTableViewCellDelegate)?) {
```

Comment:
why do we need `(UIViewController & RouteTableViewCellDelegate)?` instead of just `RouteTableViewCellDelegate?`